tools/topology/topology2/include/pipelines/cavs/gain-copier-capture.conf

```diff
@@ -0,0 +1,132 @@
+#
+# CAVS gain (volume) capture pipeline
+#
+# A simple gain (volume) pipeline. All attributes defined herein are namespaced
+# by alsatplg to "Object.Pipeline.gain-copier-capture.N.attribute_name".
+#
+# Usage: gain-copier-capture pipeline object can be instantiated as:
+#
+# Object.Pipeline.gain-copier-capture."N" {
+# 	format		"s16le"
+# 	period		1000
+# 	time_domain	"timer"
+# 	channels	2
+# 	rate		48000
+# }
+#
+# Where N is the unique pipeline ID within the same alsaconf node.
+#
+
+<include/common/audio_format.conf>
+<include/components/copier.conf>
+<include/components/gain.conf>
+<include/components/pipeline.conf>
+
+Class.Pipeline."gain-copier-capture" {
+
+	DefineAttribute."index" {}
+
+	<include/pipelines/pipeline-common.conf>
+
+	attributes {
+		!constructor [
+			"index"
+		]
+
+		!immutable [
+			"direction"
+		]
+
+		#
+		# gain-copier-capture objects instantiated within the same
+		# alsaconf node must have unique pipeline_id attribute
+		#
+		unique	"instance"
+	}
+
+	Object.Widget {
+		copier."1" {
+			copier_type	"host"
+			type	"aif_out"
+			num_audio_formats 3
+			# 16-bit 48KHz 2ch
+			Object.Base.audio_format.1 {
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				dma_buffer_size "$[$obs * 2]"
+			}
+			# 24-bit 48KHz 2ch
+			Object.Base.audio_format.2 {
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_bit_depth		32
+				out_valid_bit_depth	24
+				dma_buffer_size "$[$obs * 2]"
+			}
+			# 32-bit 48KHz 2ch
+			Object.Base.audio_format.3 {
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_bit_depth		32
+				out_valid_bit_depth	32
+				dma_buffer_size "$[$obs * 2]"
+			}
+			node_type $HDA_HOST_INPUT_CLASS
+		}
+
+		gain."1" {
+			num_audio_formats 2
+			#16-bit 48KHz 2ch
+			Object.Base.audio_format.1 {}
+			# 32-bit 48KHz 2ch
+			Object.Base.audio_format.2 {
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_bit_depth		32
+				out_valid_bit_depth	32
+			}
+		}
+
+		copier."2" {
+			type buffer
+			copier_type	"module"
+			num_audio_formats 2
+			#16-bit 48KHz 2ch
+			Object.Base.audio_format.1 {}
+			# 32-bit 48KHz 2ch
+			Object.Base.audio_format.2 {
+				in_bit_depth		32
+				in_valid_bit_depth	32
+				out_bit_depth		32
+				out_valid_bit_depth	32
+				dma_buffer_size "$[$ibs * 2]"
+			}
+		}
+
+		pipeline."1" {
+			priority	0
+			lp_mode		0
+		}
+	}
+
+	Object.Base {
+		route."1" {
+			source	"copier.module..2"
+			sink	"gain..1"
+		}
+		route."2" {
+			source	"gain..1"
+			sink	"copier.host..1"
```

Comment:
why do we have duplicate dots in the module.2, gain.1, etc. Is this a mistake or something allowed by topology2 parsers?

@ranj063 can you double-check this?

Reply:
yes this is needed so that the pipeline ID can be filled in at the top-level @plbossart